assets/source/setup-guide/app/steps/SetupAccount.js

```diff
@@ -1,117 +1,83 @@
+/* eslint-disable @wordpress/no-global-event-listener */
 /**
  * External dependencies
  */
-import { sprintf, __ } from '@wordpress/i18n';
-import { decodeEntities } from '@wordpress/html-entities';
-import { createInterpolateElement, useState } from '@wordpress/element';
-import apiFetch from '@wordpress/api-fetch';
-import { Spinner } from '@woocommerce/components';
-import { getNewPath } from '@woocommerce/navigation';
+
 import {
-	Button,
-	Card,
-	CardBody,
-	CardFooter,
-	Flex,
-	FlexItem,
-	FlexBlock,
-	Modal,
-	__experimentalText as Text, // eslint-disable-line @wordpress/no-unsafe-wp-apis --- _experimentalText unlikely to change/disappear and also used by WC Core
-} from '@wordpress/components';
+	useEffect,
+	useState,
+	useCallback,
+	createInterpolateElement,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button, Card, CardFooter, CardDivider } from '@wordpress/components';
+import apiFetch from '@wordpress/api-fetch';
 
 /**
  * Internal dependencies
  */
 import StepHeader from '../components/StepHeader';
 import StepOverview from '../components/StepOverview';
+import AccountConnection from '../components/Account/Connection';
+import BusinessAccountSelection from '../components/Account/BusinessAccountSelection';
 import { useSettingsSelect, useCreateNotice } from '../helpers/effects';
 
 const SetupAccount = ( {
 	goToNextStep,
 	view,
 	isConnected,
 	setIsConnected,
+	isBusinessConnected,
+	setIsBusinessConnected,
 } ) => {
-	const [ isConfirmationModalOpen, setIsConfirmationModalOpen ] = useState(
-		false
-	);
+	const createNotice = useCreateNotice();
 	const appSettings = useSettingsSelect();
+	const [ attemptedCreation, setAttemptedCreation ] = useState( false );
+	const [ businessAccounts, setBusinessAccounts ] = useState(
+		wcSettings.pinterest_for_woocommerce.businessAccounts
+	);
 
-	const createNotice = useCreateNotice();
+	useEffect( () => {
+		if ( attemptedCreation ) {
+			window.addEventListener( 'focus', fetchBusinesses );
+		}
 
-	const openConfirmationModal = () => {
-		setIsConfirmationModalOpen( true );
-	};
+		return () => window.removeEventListener( 'focus', fetchBusinesses );
+	}, [ fetchBusinesses, attemptedCreation ] );
 
-	const closeConfirmationModal = () => {
-		setIsConfirmationModalOpen( false );
-	};
+	useEffect( () => {
+		if ( ! isConnected ) {
+			setIsBusinessConnected( isConnected );
+		}
```

Comment:
The logic here is "if `isConnected` is false, set `isBusinessConnected` to false. Suggest to change to the following for easier reading.

```suggestion
		if ( ! isConnected ) {
			setIsBusinessConnected( false );
		}
```